metricflow/cli/custom_click_types.py

```diff
@@ -0,0 +1,119 @@
+"""This module contains custom helper click types."""
+
+from typing import Any, Callable, Dict, Generic, List, Optional, TypeVar
+import click
+
+T = TypeVar("T")
+
+
+class ListParamType(click.ParamType, Generic[T]):
```

Comment:
One issue that I found with `List` is that it's not covariant, so I've generally been trying to use `Sequence` or `Tuple`.